autoemulate/calibration/history_matching.py

```diff
@@ -319,83 +320,113 @@ def __init__(  # noqa: PLR0913 allow too many arguments since all currently requ
         self.emulator = emulator
         self.emulator.device = self.device
 
-        # These get updated when run() is called and used to refit the emulator
+        # These get updated when `run()` is called and used to refit the emulator
         if train_x is not None and train_y is not None:
             self.train_x = train_x.float().to(self.device)
             self.train_y = train_y.float().to(self.device)
         else:
             self.train_x = torch.empty((0, self.simulator.in_dim), device=self.device)
             self.train_y = torch.empty((0, self.simulator.out_dim), device=self.device)
 
-    def generate_samples(self, n: int) -> tuple[TensorLike, TensorLike]:
+        # Store NROY samples geneated in `run()`, use in cloud sampling at next wave
+        self.nroy_samples = None
+
+    def cloud_sample(self, n: int, scaling_factor: float = 0.1) -> TensorLike:
```

Comment:
In GPerks they implement further logic that excludes any sampled points which are outside the expected parameter bounds - we probably want to do something similar.